From fb8f1438a40668a566ac19b55ebfbf691142893e Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Sat, 27 Nov 2021 12:18:17 +0100
Subject: [PATCH] lxc.py: Detect usage of VM and set correct gralloc value

When using Qemu/VirtualBox, gralloc is set to "gbm" while it should be "default". Add py_vmdetect to detect if we're running in a virtual machine and set gralloc accordingly.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 tools/helpers/lxc.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tools/helpers/lxc.py b/tools/helpers/lxc.py
index d57e63c..c0578a5 100644
--- a/tools/helpers/lxc.py
+++ b/tools/helpers/lxc.py
@@ -9,6 +9,7 @@
 import platform
 import tools.config
 import tools.helpers.run
+from py_vmdetect import VMDetect
 
 
 def get_lxc_version(args):
@@ -168,11 +169,15 @@ def find_hal(hardware):
     props = []
     egl = tools.helpers.props.host_get(args, "ro.hardware.egl")
 
+    vmd = VMDetect()
     gralloc = find_hal("gralloc")
     if gralloc == "":
         if os.path.exists("/dev/dri"):
-            gralloc = "gbm"
             egl = "mesa"
+            if vmd.is_vm():
+                gralloc = "default"
+            else:
+                gralloc = "gbm"
         else:
             gralloc = "default"
             egl = "swiftshader" 
